final/tool/menus.cpp

```diff
@@ -0,0 +1,192 @@
+#include "stdafx.h"
+#include <ostream>
+#include <iostream>
+#include <string>
+#include <sstream>
+#include <fstream>
+#include <map>
+#include <vector>
+#include "block.h"
+#include <algorithm>
+#include "menus.h"
+
+void press_enter_to_go_back()
+{
+  std::cout << "\nPress ENTER to go back...\n";
+  std::cin.ignore();
+  system("cls");
+}
+
+unsigned int get_number_from_console(std::string caption)
+{
+  unsigned int user_number = 0;
+  std::string input = "";
+  while (true) {
+    std::cout << caption;
+    getline(std::cin, input);
+    // This code converts from string to number safely.
+    std::stringstream input_ss(input);

```

Comment:
https://en.cppreference.com/w/cpp/string/basic_string/stol
https://en.cppreference.com/w/cpp/string/basic_string/stof

Reply:
with current code I have 'try' approach instead of handling exception